src/routes/api/__tests__/artists.js

```diff
@@ -0,0 +1,187 @@
+const supertest = require('supertest');
+const { format } = require('date-fns'); // Mencionar esta librería
+const app = require('../../../app');
+
+const request = supertest(app.callback());
+
+describe('Artist API routes', () => {
+  beforeAll(async () => {
+    await app.context.orm.sequelize.sync({ force: true });
+  });
+
+  afterAll(async () => {
+    await app.context.orm.sequelize.close();
+  });
+
+  describe('GET /api/artists/:id', () => {
+    let artist;
+    let response;
+    const artistData = {
+      name: 'Imagine Dragons',
+      origin: 'Las Vegas, Nevada, United States',
+      genres: 'Rock, Reggae, Alternative/Indie',
+      foundedIn: 2009,
+      members: 'Dan Reynolds, Ben McKee, Daniel Wayne Sermon, Daniel Platzman',
+    };
+
+    const getArtist = (id) => request.get(`/api/artists/${id}`);
+
+    beforeAll(async () => {
+      artist = await app.context.orm.artist.create(artistData);
+    });
+
+    describe('When passed id corresponds to an existing artist', () => {
+      beforeAll(async () => {
+        response = await getArtist(artist.id);
+      });
+
+      test('responds with 200 status code', () => {
+        expect(response.status).toBe(200);
+      });
+
+      test('response attributes match saved artist', () => {
+        expect(response.body.data.attributes).toEqual(artistData);
+      });
+
+      test('body matches snapshot', () => {
+        expect(response.body).toMatchSnapshot();
+      });
+    });
+
+    describe('When passed id does not correspond to any author', () => {
+      test('responds with 404 status code', async () => {
+        response = await getArtist(artist.id * -1);
+        expect(response.status).toBe(404);
+      });
+    });
+  });
+
+  const artistData = {
+    name: 'Unknown Mortal Orchestra',
+    origin: 'Auckland, New Zealand',
+    genres: 'Lofi, Psychedelic Rock, Alternative/Indie',
+    foundedIn: 2009,
+    members: 'Ruban Nielson, Jake Portrait, Kody Nielson',
+  };
+
+  const userData = {
+    firstName: 'Jane',
+    lastName: 'Doe',
+    email: 'janedoe@gmail.com',
+    password: 'web1234',
+  };
+  const { email, password } = userData;
+
+  const postAuth = (body) => request
+    .post('/api/auth')
+    .set('Content-type', 'application/json')
+    .send(body);
+
+  describe('POST /api/artists', () => {
+    let response;
+
+    beforeAll(async () => {
+      await app.context.orm.user.create(userData);
+    });
+
+    let token;
+
+    let createArtist = () => request.post('/api/artists');
+    describe('When user is not authorized ❌', () => {
+      beforeAll(async () => {
+        response = await createArtist(artistData);
+      });
+      it('should return a 401 status code', () => {
```

Comment:
Creo que me opongo a usar `it` en este archivo particularmente, dado que ya se vió en la ayudantía con `test` y creo que puede llegar a confundir a los estudiantes. Lo usaría para mostrar una alternativa a `test`, pero no en este archivo.

Reply:
La verdad es que estoy en desacuerdo. Si ya mostré el `it` antes (lo primero que se ve son los test de artist), ya deberían entender los dos. Creo que mantenerse con el `test` "porque podría confundirlos" es sobre protegerlos jajaj